lib/src/set.dart

```diff
@@ -263,12 +272,24 @@ class ManagedRealmSet<T extends Object?> with RealmEntity, SetMixin<T> implement
 
   @override
   RealmResults<T> asResults() => RealmResultsInternal.create<T>(realmCore.resultsFromSet(this), realm, _metadata);
+
+  @override
+  RealmSet<T> freeze() {
+    if (isFrozen) {
+      return this;
+    }
+
+    final frozenRealm = realm.freeze();
+    return frozenRealm.resolveSet(this)!;
```

Comment:
I think if we return `RealmError` when `resolveSet` returns `null`, we will have better traceability than if we insist that the returned value is not null with "!". I mean that it is better to have a specific error than a common error. I hope this shouldn't happen, so that you can leave it like this if you decide.

Reply:
No, it should never return null in this case, since we are just resolving in the frozen version of the current realm. Same goes for list and object.